// @compositor/kit/dist/Library.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Detail = exports.Example = exports.Library = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _class, _temp2;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _nanoStyle = require('nano-style');

var _nanoStyle2 = _interopRequireDefault(_nanoStyle);

var _reactRouterDom = require('react-router-dom');

var _ui = require('./ui');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Root = (0, _nanoStyle2.default)('div')({
  display: 'flex',
  alignItems: 'flex-start',
  height: '100vh'
});

var Main = (0, _nanoStyle2.default)('div')({
  flex: '1 1 auto',
  height: '100vh',
  overflowY: 'auto',
  WebkitOverflowScrolling: 'touch'
});

var Card = (0, _nanoStyle2.default)(_reactRouterDom.Link)({
  display: 'block',
  color: 'inherit',
  textDecoration: 'none',
  overflow: 'hidden',
  border: '1px solid #f6f6f6',
  '&:hover': {
    borderColor: '#ddd'
  }
});

var SideBar = (0, _nanoStyle2.default)('div')({
  width: '192px',
  flex: 'none',
  height: '100vh',
  overflowY: 'auto',
  WebkitOverflowScrolling: 'touch',
  borderRight: '1px solid #f6f6f6'
});

var NavItem = (0, _nanoStyle2.default)(_reactRouterDom.NavLink)({
  display: 'block',
  paddingLeft: '8px',
  paddingRight: '8px',
  paddingTop: '4px',
  paddingBottom: '4px',
  fontSize: '12px',
  fontWeight: 'bold',
  textDecoration: 'none',
  color: 'inherit',
  WebkitUserSelect: 'none',
  userSelect: 'none',
  '&.active': {
    color: 'white',
    backgroundColor: 'black'
  }
});

var Router = typeof document !== 'undefined' ? _reactRouterDom.BrowserRouter : _reactRouterDom.StaticRouter;

var Library = exports.Library = function (_React$Component) {
  _inherits(Library, _React$Component);

  function Library() {
    _classCallCheck(this, Library);

    return _possibleConstructorReturn(this, (Library.__proto__ || Object.getPrototypeOf(Library)).apply(this, arguments));
  }

  _createClass(Library, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          basename = _props.basename,
          props = _objectWithoutProperties(_props, ['basename']);

      return _react2.default.createElement(
        Router,
        { basename: basename, context: {} },
        _react2.default.createElement(LibraryApp, props)
      );
    }
  }]);

  return Library;
}(_react2.default.Component);

var LibraryApp = (0, _reactRouterDom.withRouter)((_temp2 = _class = function (_React$Component2) {
  _inherits(_class, _React$Component2);

  function _class() {
    var _ref;

    var _temp, _this2, _ret;

    _classCallCheck(this, _class);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this2 = _possibleConstructorReturn(this, (_ref = _class.__proto__ || Object.getPrototypeOf(_class)).call.apply(_ref, [this].concat(args))), _this2), _this2.getExampleChildren = function (_ref2) {
      var children = _ref2.children;
      return _react2.default.Children.toArray(children).filter(function (c) {
        return c.type === Example;
      }).filter(function (c) {
        return !!c.props.name;
      }).map(function (c) {
        return {
          name: c.props.name,
          element: c.props.children
        };
      });
    }, _temp), _possibleConstructorReturn(_this2, _ret);
  }

  _createClass(_class, [{
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          title = _props2.title,
          renderSideNav = _props2.renderSideNav,
          renderCard = _props2.renderCard;


      var examples = this.props.examples || this.getExampleChildren(this.props);

      var sidenav = typeof renderSideNav === 'function' ? renderSideNav(_extends({}, this.state, { title: title, examples: examples })) : _react2.default.createElement(SideNav, { title: title, examples: examples });

      return _react2.default.createElement(
        Root,
        null,
        _react2.default.createElement(
          SideBar,
          null,
          sidenav
        ),
        _react2.default.createElement(
          Main,
          null,
          _react2.default.createElement(_reactRouterDom.Route, {
            exact: true,
            path: '/',
            render: function render() {
              return _react2.default.createElement(
                _ui.Grid,
                null,
                examples.map(function (example) {
                  return _react2.default.createElement(
                    Card,
                    {
                      key: example.name,
                      to: '/' + example.name },
                    typeof renderCard === 'function' ? renderCard(example) : _react2.default.createElement(
                      _ui.Box,
                      { p: 2 },
                      example.element
                    )
                  );
                })
              );
            }
          }),
          examples.map(function (example) {
            return _react2.default.createElement(_reactRouterDom.Route, {
              key: example.name,
              path: '/' + example.name,
              render: function render() {
                return _react2.default.createElement(
                  _ui.Box,
                  { p: 2 },
                  example.element
                );
              }
            });
          })
        )
      );
    }
  }]);

  return _class;
}(_react2.default.Component), _class.propTypes = {
  title: _propTypes2.default.string,
  examples: _propTypes2.default.array,
  renderSideNav: _propTypes2.default.func,
  renderCard: _propTypes2.default.func
}, _temp2));

var SideNav = function (_React$Component3) {
  _inherits(SideNav, _React$Component3);

  function SideNav() {
    _classCallCheck(this, SideNav);

    return _possibleConstructorReturn(this, (SideNav.__proto__ || Object.getPrototypeOf(SideNav)).apply(this, arguments));
  }

  _createClass(SideNav, [{
    key: 'render',
    value: function render() {
      var _props3 = this.props,
          examples = _props3.examples,
          title = _props3.title;


      return _react2.default.createElement(
        _react2.default.Fragment,
        null,
        _react2.default.createElement(
          NavItem,
          { exact: true, to: '/' },
          title
        ),
        examples.map(function (example) {
          return _react2.default.createElement(
            NavItem,
            {
              key: example.name,
              to: '/' + example.name },
            example.name
          );
        })
      );
    }
  }]);

  return SideNav;
}(_react2.default.Component);

SideNav.propTypes = {
  title: _propTypes2.default.string,
  examples: _propTypes2.default.array
};
SideNav.defaultProps = {
  title: 'Kit',
  examples: []
};

var Example = exports.Example = function (_React$Component4) {
  _inherits(Example, _React$Component4);

  function Example() {
    _classCallCheck(this, Example);

    return _possibleConstructorReturn(this, (Example.__proto__ || Object.getPrototypeOf(Example)).apply(this, arguments));
  }

  _createClass(Example, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(_react2.default.Fragment, this.props);
    }
  }]);

  return Example;
}(_react2.default.Component);

Example.propTypes = {
  name: _propTypes2.default.string.isRequired
};
var Detail = exports.Detail = (0, _reactRouterDom.withRouter)(function (_React$Component5) {
  _inherits(_class2, _React$Component5);

  function _class2() {
    _classCallCheck(this, _class2);

    return _possibleConstructorReturn(this, (_class2.__proto__ || Object.getPrototypeOf(_class2)).apply(this, arguments));
  }

  _createClass(_class2, [{
    key: 'render',
    value: function render() {
      var location = this.props.location;

      if (location.pathname === '/') return false;

      return _react2.default.createElement('div', this.props);
    }
  }]);

  return _class2;
}(_react2.default.Component));

exports.default = Library;
