// @compositor/kit/dist/Responsive.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _nanoStyle = require('nano-style');

var _nanoStyle2 = _interopRequireDefault(_nanoStyle);

var _Frame = require('./Frame');

var _Frame2 = _interopRequireDefault(_Frame);

var _ui = require('./ui');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Root = (0, _nanoStyle2.default)(_ui.Flex)({
  overflowX: 'auto'
});

var Card = (0, _nanoStyle2.default)(_ui.Box)({
  border: '1px solid #f6f6f6'
});

var Label = (0, _nanoStyle2.default)('div')({
  fontSize: '10px'
});

var Responsive = function (_React$Component) {
  _inherits(Responsive, _React$Component);

  function Responsive() {
    _classCallCheck(this, Responsive);

    return _possibleConstructorReturn(this, (Responsive.__proto__ || Object.getPrototypeOf(Responsive)).apply(this, arguments));
  }

  _createClass(Responsive, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          viewports = _props.viewports,
          zoom = _props.zoom,
          children = _props.children;


      return _react2.default.createElement(
        Root,
        null,
        viewports.map(function (viewport) {
          return _react2.default.createElement(
            _ui.Box,
            { key: viewport.width + 'x' + viewport.height, p: 3 },
            _react2.default.createElement(
              Card,
              null,
              _react2.default.createElement(_Frame2.default, _extends({}, viewport, { zoom: zoom, children: children }))
            ),
            _react2.default.createElement(
              Label,
              null,
              viewport.width,
              ' x ',
              viewport.height,
              ' @',
              zoom,
              'x'
            )
          );
        })
      );
    }
  }]);

  return Responsive;
}(_react2.default.Component);

Responsive.defaultProps = {
  viewports: [{ width: 375, height: 667 }, { width: 414, height: 736 }, { width: 1024, height: 768 }, { width: 1366, height: 1024 }]
};
exports.default = Responsive;
