// standard-components/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = exports.Button = exports.BackgroundImage = exports.Image = exports.Heading = exports.Text = exports.Flex = exports.Box = undefined;

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

var _cleanTag = require('clean-tag');

var _cleanTag2 = _interopRequireDefault(_cleanTag);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Box = exports.Box = (0, _styledComponents2.default)(_cleanTag2.default)([], _styledSystem.space, _styledSystem.color, _styledSystem.width, _styledSystem.fontSize, _styledSystem.flex, _styledSystem.alignSelf, _styledSystem.order);

Box.displayName = 'Box';

var Flex = exports.Flex = (0, _styledComponents2.default)(Box)([], {
  display: 'flex'
}, _styledSystem.alignItems, _styledSystem.justifyContent, _styledSystem.flexDirection, _styledSystem.flexWrap);

Flex.displayName = 'Flex';

var Text = exports.Text = (0, _styledComponents2.default)(_cleanTag2.default)([], _styledSystem.space, _styledSystem.color, _styledSystem.fontSize, _styledSystem.fontWeight, _styledSystem.lineHeight, _styledSystem.textAlign, _styledSystem.letterSpacing);

Text.displayName = 'Text';

var Heading = exports.Heading = (0, _styledComponents2.default)(Text)([]);

Heading.displayName = 'Heading';

Heading.defaultProps = {
  is: 'h2',
  fontSize: 4,
  fontWeight: 'bold',
  lineHeight: 1.25
};

var Image = exports.Image = (0, _styledComponents2.default)(_cleanTag2.default.img)([], {
  display: 'block',
  maxWidth: '100%',
  height: 'auto'
}, _styledSystem.space, _styledSystem.color, _styledSystem.width);

Image.displayName = 'Image';

var BackgroundImage = exports.BackgroundImage = (0, _styledComponents2.default)(Box)([], _styledSystem.backgroundImage, _styledSystem.backgroundSize, _styledSystem.backgroundPosition);

BackgroundImage.displayName = 'BackgroundImage';

BackgroundImage.defaultProps = {
  backgroundSize: 'cover',
  backgroundPosition: 'center'
};

var Button = exports.Button = (0, _styledComponents2.default)(_cleanTag2.default.button)([], {
  fontFamily: 'inherit',
  fontWeight: 'bold',
  lineHeight: 1.25,
  appearance: 'none',
  textDecoration: 'none',
  display: 'inline-block',
  verticalAlign: 'middle',
  textAlign: 'center'
}, _styledSystem.space, _styledSystem.color, _styledSystem.fontSize, _styledSystem.border, _styledSystem.borderRadius, _styledSystem.borderColor);

Button.displayName = 'Button';

Button.defaultProps = {
  m: 0,
  px: 3,
  py: 2,
  fontSize: 'inherit',
  color: 'white',
  bg: 'blue',
  border: 0,
  borderRadius: 2
};

var Link = exports.Link = (0, _styledComponents2.default)(_cleanTag2.default.a)([], _styledSystem.space, _styledSystem.color);

Link.displayName = 'Link';

exports.default = {
  Box: Box,
  Flex: Flex,
  Text: Text,
  Heading: Heading,
  Image: Image,
  BackgroundImage: BackgroundImage,
  Button: Button,
  Link: Link
};
